icu4j/main/core/src/test/java/com/ibm/icu/dev/test/message2/TestUtils.java

```diff
@@ -134,10 +173,25 @@ static Reader jsonReader(String jsonFileName) throws URISyntaxException, IOExcep
         return Files.newBufferedReader(json, StandardCharsets.UTF_8);
     }
 
-    private static Path getTestFile(Class<?> cls, String fileName) throws URISyntaxException {
+    private static Path getTestFile(Class<?> cls, String fileName) throws URISyntaxException, IOException {
         String packageName = cls.getPackage().getName().replace('.', '/');
         URI getPath = cls.getClassLoader().getResource(packageName).toURI();
         Path filePath = Paths.get(getPath);
         Path json = Paths.get(fileName);
-        return filePath.resolve(json);
+        // First, check the top level of the source directory,
+        // in case we're in a source tarball
```

Comment:
Same question as for ICU4C: Should we start to package up ICU4J zip/jar/tar files from repo root, except omit unnecessary folders?